internal/runner/accept.go

```diff
@@ -22,6 +22,31 @@ import (
 
 var heartbeatDuration = 5 * time.Second
 
+// AcceptParallel allows up to count jobs to be accepted and executing
+// concurrently.
+func (r *Runner) AcceptParallel(ctx context.Context, count int) {
+	// Create a new cancellable context so we can stop all the goroutines
+	// when one exits. We do this because if one exits, its likely that the
+	// unrecoverable error exists in all.
+	ctx, cancel := context.WithCancel(ctx)
+	defer cancel()
+
+	// Start up all the goroutines
+	r.logger.Info("accepting jobs concurrently", "count", count)
+	var wg sync.WaitGroup
+	wg.Add(count)
+	for i := 0; i < count; i++ {
+		go func() {
+			defer cancel()
```

Comment:
Just saying this out loud, should the exit of ANY of the goroutines cause them all to exit?

Reply:
So actually, yes I think this needs to do slightly better, although its edge case-y. Here is the thought:

1. If the user cancels the context, then this is effectively a no-op and all of the goroutines are canceled anyways. No issue.

2. If a goroutine has an error, its likely the error will impact all, because there is _considerable_ retry logic already in each `Accept` call -- including reconnection -- so if it actually errors it is likely unrecoverable. So we DO want to exit all the goroutines.

_However_, for #2, right now we're canceling the context which just causes a cascade effect to cancel ASAP. I think we can do better by just letting each existing job _try_ to finish gracefully, and then say "don't accept any more jobs thereafter." 

I'll work on this Monday.

Reply:
Actually, I think this is okay for now. Its a bit non-trivial to get this fix in and looking at the possible reasons for a return from `AcceptMany`, i do think things are really broken if they exit so cancelling all is okay for now. We can improve this later. I've added a TODO to note it.